### DISCUSSION TITLE
Display, interact and edit multiple point features with different image sources on WebGLPointsLayer

### DISCUSSION THREAD

**Discussion by nebsar** [openlayers/openlayers#14623] [Q&A]
I would like to display multiple point features with different image sources on WebGLPointsLayer as the screenshot below which is possible in leaflet PixiOverlay. And I would like to dynamically change the image source of a single point. Is this possible?

![image](https://user-images.githubusercontent.com/5169314/228565326-e1f6ec91-7170-4e66-b339-8dfb008e1d17.png)

**Comment by jahow** [comment]
It's possible if you use a sprite sheet for the icons. The webgl points layer currently does not support dynamic expressions for the `src` attribute of the symbol style.

**Comment by nebsar** [comment]
It is not possible for me to use a sprite sheet, since there are thousands of different icon images.  When I use VectorLayer the web page gets sluggish with thousands of icons. Is there any plan for implementing dynamic expressions for the src attribute of the symbol style?

**Reply by jahow** [reply]
No concrete plan that I know of, even though this feature has been requested a few times (https://github.com/openlayers/openlayers/issues/11133, https://github.com/openlayers/openlayers/issues/14310, https://github.com/openlayers/openlayers/issues/12867).

It is theoretically doable, but if you really have > 1000 different icons then it might become an issue anyway I think. How many features do you typically have on the map?

**Comment by nebsar** [comment]
5K-10K icons. Some times even higher numbers. Once I needed 50K icons. Clustering may be solution for this. But user wants to disable clustering sometimes.
